File: 92.py
from typing import Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def reverseBetween(self, head: Optional[ListNode], left: int, right: int) -> Optional[ListNode]:
        newHead = ListNode(0, head)
        current = newHead
        difference = right - left

        for i in range(left-1):
            current = current.next

        while difference != 0:
            self.sendRight(current, difference)
            difference -= 1

        return newHead.next
    
    def sendRight(self, node, steps):
        next = node.next

        for i in range(steps):
            if next.next == None:
                return

            node.next = next.next
            next.next = next.next.next
            node.next.next = next
            node = node.next
            next = node.next


sol = Solution()
a = ListNode(5)
b = ListNode(4, a)
c = ListNode(3, b)
d = ListNode(2, c)
e = ListNode(1, d)

sol.reverseBetween(b, 1, 2)

File: javascript150/92.ts

class ListNode {
    val: number
    next: ListNode | null
    constructor(val?: number, next?: ListNode | null) {
        this.val = (val===undefined ? 0 : val)
        this.next = (next===undefined ? null : next)
    }
}


function reverseBetween(head: ListNode | null, left: number, right: number): ListNode | null {
    let newHead = new ListNode(0, head)
    let lp = newHead, rp = newHead

    for (let i = 0; i < right-left+1; i++) {
        rp = rp!.next!
    }

    for (let i = 0; i < left-1; i++) {
        lp = lp!.next!
        rp = rp!.next!
    }

    for (let i = 0; i < right-left; i++) {
        let next = lp!.next
        lp!.next = next!.next

        next!.next = rp!.next
        rp!.next = next
    }

    return newHead.next
};

let a = new ListNode(5)
let b = new ListNode(4, a)
let c = new ListNode(3, b)
let d = new ListNode(2, c)
let e = new ListNode(1, d)

reverseBetween(b, 1, 2)